test/test_nn.py

```diff
@@ -3573,7 +3573,11 @@ def add_test(test):
     if hasattr(TestNN, cuda_test_name):
         raise RuntimeError('Found two tests with the same name: ' + cuda_test_name)
     setattr(TestNN, test_name, lambda self, test=test: test(self))
-    setattr(TestNN, cuda_test_name, lambda self, test=test: test.test_cuda(self))
+    # Hardshrink is not implemented in CUDA, so we must not test it.
+    # TODO: Figure out why this codepath is being executed; we have another
+    # test for this already in the file
```

Comment:
The reason it succeeded before is because it catches `NotImplementedError` and treats that as a success. Now, F.hardshrink on a CUDA tensor raises a `RuntimeError`

Reply:
OK I'm just going to expunge this comment.